core/backend/src/LocalWebViewerHost.ts

```diff
@@ -0,0 +1,44 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module IModelHost
+ */
+
+import * as ws from "ws";
+import { IpcWebSocket, IpcWebSocketBackend, IpcWebSocketMessage, IpcWebSocketTransport } from "@bentley/imodeljs-common";
+import { IpcHost } from "./IpcHost";
+
+class LocalTransport extends IpcWebSocketTransport {
+  private _server: ws.Server;
+  private _connection: ws | undefined;
+
+  public constructor(port: number) {
+    super();
+
+    this._server = new ws.Server({ port });
+
+    this._server.on("connection", (connection) => {
+      this._connection = connection;
+
+      this._connection.on("message", (data) => {
+        for (const listener of IpcWebSocket.receivers)
+          listener({} as Event, JSON.parse(data as string));
+      });
+    });
+  }
+
+  public send(message: IpcWebSocketMessage): void {
+    this._connection!.send(JSON.stringify(message));
```

Comment:
what about `ArrayBuffer`s?

Reply:
I need to port the whole websocket world from mobile for that...this was just enough for the DTA case for now.

Reply:
OK, I don't think any of the current IpcApp functions use arraybuffers so it should be fine. Probably won't be any soon unless @pmconne wants to use it for temporary graphics (maybe images?)

Reply:
No immediate requirements for that.

Reply:
but I do think DTA will rely on host -> app messages for the editing stuff. I'm sure we can just use Electron for that, so probably can wait for another PR.